code/game/objects/items/stacks/tiles/tile_types.dm

```diff
@@ -1,51 +1,109 @@
-/* Diffrent misc types of tiles
-* Contains:
-*		Grass
-*		Wood
-*		Carpet
-*/
-
-/*
-* Grass
-*/
-/obj/item/stack/tile/grass
-	name = "grass tile"
-	singular_name = "grass floor tile"
-	desc = "A patch of grass like they often use on golf courses."
-	icon_state = "tile_grass"
+/obj/item/stack/tile
+	name = "floor tile"
+	singular_name = "floor tile"
+	desc = "Those could work as a pretty decent throwing weapon"
+	icon_state = "tile"
 	w_class = WEIGHT_CLASS_NORMAL
 	force = 1
 	throwforce = 1
 	throw_speed = 5
-	throw_range = 20
+	throw_range = 9
 	max_amount = 60
+	///The turf type this tile creates
+	var/turf/open/floor/turf_type
+
+/obj/item/stack/tile/Initialize(mapload)
+	. = ..()
+	pixel_x = rand(1, 14)
+	pixel_y = rand(1, 14)
+
+/obj/item/stack/tile/attack_turf(turf/T, mob/living/user)
+	if(!turf_type)
+		return
+	if(T.type != /turf/open/floor/plating)
```

Comment:
Why aren't these checks istype?

Reply:
funny old code. I moved a lot of old shit around.